pymeasure/instruments/newport/conexcc.py

```diff
@@ -0,0 +1,147 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class ConexCC(Instrument):
+    """
+    Class for Newport ConexCC Controllers
+    """
+    def __init__(self, address, index=1, **kwargs):
+        super(ConexCC, self).__init__(
+            address, "Conex-CC Controller for Linear Stages", baud_rate=921600,
+            read_termination='\r\n', write_termination='\r\n', **kwargs)
+        # Implementation of this index should allow for communication with
+        # serially connected controllers on a single USB interface
+        # This is not (yet) tested however!
+        self.index = index
+        self.zero()
+
+    def ask(self, command):
+        super(ConexCC, self).ask(str(self.index)+command)
+
+    def write(self, command):
+        super(ConexCC, self).write(str(self.index)+command)
+
+    def values(self, command, **kwargs):
+        # Cut off the repetition of the command that is included in the reply
+        reply = super(ConexCC, self).values(
+            str(self.index)+command, **kwargs)[0][len(str(self.index))+2:]
+        return reply
+
+    acceleration = Instrument.control(
+        'AC?', 'AC%f',
+        """ Acceleration of the stage in
+        units/s^2. """,
+        get_process=lambda v: float(v),
+        validator=strict_range,
+        values=(1e-6, 1e12))
+
+    velocity = Instrument.control(
+        'VA?', 'VA%f',
+        """ Acceleration of the stage in
+        units/s. """,
+        get_process=lambda v: float(v),
+        validator=strict_range,
+        values=(1e-6, 1e12))
+
+    position = Instrument.measurement(
+        'TP', """ Get current Stage Position. """)
+
+    Kp = Instrument.control(
+        'KP?', 'KP%f',
+        """ Proportional Gain for PID control loop.""",
+        validator=strict_range,
+        values=(0, 1e12)
+    )
+
+    Ki = Instrument.control(
+        'KI?', 'KI%f',
+        """ Integral Gain for PID control loop.""",
+        validator=strict_range,
+        values=(0, 1e12)
+    )
+
+    Kd = Instrument.control(
+        'KD?', 'KD%f',
+        """ Derivative Gain for PID control loop.""",
+        validator=strict_range,
+        values=(0.001, 1e12)
+    )
+
+    PID_enabled = Instrument.control(
+        'SC?', 'SC%g',
+        """ PID Control Loop State (enabled/disabled).""",
+        map_values=True, values={True: 1, False: 0},
+        validator=strict_discrete_set
+    )
+
+    def move_to(self, pos):
+        """ Move to absolute position
+        :param pos Absolute position to move to.
+        """
+        cmd = 'PA%f' % (pos)
+        self.write(cmd)
```

Comment:
`self.write(f'{cmd:f}') should do the same thing and is a more modern string formatting approach. Similar for other occurences.